standards/application/messaging-api.md

```diff
@@ -0,0 +1,944 @@
+}
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "subscriptionId": "9978c4f1-4465-4537-9d95-0cf016fb1080"
+}
+```
+
+Error for `404`:
+```http
+HTTP/1.1 404 Not Found
+Content-Type: application/json
+
+{
+  "error": "Cannot create subscription. clusterId 0 is not supported."
+}
+```
+
+##### `unsubscribe`
+Request:
+```http
+POST /unsubscribe HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+
+{
+  "subscriptionId": "9978c4f1-4465-4537-9d95-0cf016fb1080"
+}
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "status": "ok"
+}
+```
+
+##### `getActiveSubscriptions`
+Request:
+```http
+GET /subscriptions HTTP/1.1
+Accept: application/json
+```
+
+Response:
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+[
+  "9978c4f1-4465-4537-9d95-0cf016fb1080",
+  "a1b2c3d4-5678-90ab-cdef-1234567890ab"
+]
+```
+
+### Message Storage
+
+The `Message Storage API` SHOULD be used to retrieve information about messages that have been sent, 
+received, or stored.
+
+Message storage is populated by:
+- Messages sent by the underlying node via the `Send API`.
+- Messages received through an active subscription initiated by the `Subscribe API`.
+- Messages fetched using the `History API`.
+
+Messages can be identified by:
+- The `message hash` as described in the [MESSAGE](https://github.com/vacp2p/rfc-index/blob/8ee2a6d6b232838d83374c35e2413f84436ecf64/waku/standards/core/14/message.md#deterministic-message-hashing) specification.
+- The `contentTopic` associated with the message, as defined in the [TOPICS](https://github.com/vacp2p/rfc-index/blob/main/waku/informational/23/topics.md#content-topic-format) specification.
+- The `RequestID` described in the `Send API` section.
+- The `SubscriptionID` described in the `Subscribe API` section.
+
+These identifiers SHOULD be passed to the appropriate methods described below to retrieve or manage messages.
+
+Depending on the presence of the `storeConfirmation` or `receiveConfirmation` options in the `Initial configuration`, the `Message Storage API` will trigger either the `Subscribe API` or `Acknowledgements` routine.
+
+Once a message is added to the `Message Storage API`, or if any details about it change,
+the appropriate events from the `message:*` event family (as described in the `Event Source API` section) MUST be invoked after the message is added or updated in the underlying storage.
+
+#### Acknowledgements
+
+Message Acknowledgement mechanism is a background operation that MUST:
+- Perform store‑based reliability checks as specified in [P2P‑RELIABILITY](./p2p‑reliability.md).  
+- Halt when the node’s health status transitions to `unhealthy` (as defined by the `Health API`).  
+- Resume when the node’s health status transitions to either `minimally healthy` or `healthy`.  
+- Set the `stored` or `received` field on `MessageRecord` once the message is acknowledged according to store‑based reliability.
```

Comment:
I would not use the term `acknowledgements` as it implies e2e reliability.

I would not list all of those and leave it to implementation detail. just refer to p2p reliability.